Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -125,60 +129,80 @@ namespace GradientSignal
                 // imageBoundsY)
                 AZ::Vector3 uvw(static_cast<float>(x), static_cast<float>((imageResolutionY - 1) - y), 0.0f);
 
-                GradientSampleParams sampleParams;
-                sampleParams.m_position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
+                AZ::Vector3 position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
 
                 bool inBounds = true;
                 LmbrCentral::ShapeComponentRequestsBus::EventResult(
-                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, sampleParams.m_position);
+                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, position);
```

Comment:
This loop would be significantly faster if you turn it into a single EBus call with a lambda. Here's an example of what that looks like in GradientSurfaceDataComponent.cpp:
```
            LmbrCentral::ShapeComponentRequestsBus::Event(
                m_configuration.m_shapeConstraintEntityId,
                [positions, shapeConstraintBounds, &inBounds](LmbrCentral::ShapeComponentRequestsBus::Events* shape)
                {
                    inBounds.resize(positions.size(), false);

                    for (size_t index = 0; index < positions.size(); index++)
                    {
                        // Check the AABB first.
                        if (shapeConstraintBounds.Contains(positions[index]))
                        {
                            // The point is in the AABB, so check against the actual shape geometry.
                            inBounds[index] = shape->IsPointInside(positions[index]);
                        }
                    }
                });
    ```

Reply:
Good call, will implement!